src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpUseAssignableTypeForForeachVariable.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.NetCore.Analyzers.Runtime;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpUseAssignableTypeForForeachVariable : UseAssignableTypeForForeachVariable
+    {
+        protected override void AnalyzeLoop(OperationAnalysisContext context, Compilation compilation)
```

Comment:
Open question: Just wondering if it would be worth transferring some of the logic in the base class and having the subclass only proving some info (the syntax, the `GetForEachStatementInfo`).

Reply:
`ForEachStatementInfo` is a C# class and an entire `AnalyzeLoop` method depends on it

Reply:
You are right, that will force adding too many methods on base.